Do not retry "invalid ticket senderNonce" for Live Video to Video

leszko opened the pull request:
The "invalid ticket senderNonce" should not be retries or Live Video to Video, because retrying it does not help. The session needs to be refreshed to retry it successfully

leszko committed:
The "invalid ticket senderNonce" should not be retries or Live Video to Video, because retrying it does not help. The session needs to be refreshed to retry it successfully

leszko commented:
@ad-astra-video I discovered that retrying this "invalid ticket nonces" error does not work, because a retry results in the same error. Therefore we need to skip the give O. I wonder what should be the behavior for the AI Jobs?

leszko committed:
Fix unit tests

ad-astra-video commented:
I am ok with adding same behavior for batch jobs as well.  

If a invalid nonce error happens its likely due to pricing issues on the request for that Orchestrator that is not likely to update in the request timeout.

leszko committed:
Change the logic for Live Video to Video

leszko committed:
Change the logic for Live Video to Video

leszko commented:
> I am ok with adding same behavior for batch jobs as well.
> 
> If a invalid nonce error happens its likely due to pricing issues on the request for that Orchestrator that is not likely to update in the request timeout.

Ok, updated to use the same logic for AI Batch and Live Video to Video.

I've changed it slightly to not return the sessions to `retryableSessions`, because we need the session refresh before we this nonces error disappear. Please take a look if it work for you. If not please send a PR with the changed logic for AI Jobs.

leszko committed:
Merge branch 'master' into rafal/dont-retry-invalid-nonces
